discovery/discovery.go

```diff
@@ -48,7 +51,7 @@ func NewOrchestratorPool(bcast common.Broadcaster, uris []*url.URL, score float3
 func NewOrchestratorPoolWithPred(bcast common.Broadcaster, addresses []*url.URL,
 	pred func(*net.OrchestratorInfo) bool, score float32, orchBlacklist []string) *orchestratorPool {
 
-	pool := NewOrchestratorPool(bcast, addresses, score, orchBlacklist)
+	pool := NewOrchestratorPool(bcast, addresses, score, orchBlacklist, 0) //last parameter is timeout that is set at launch, 0 keeps same setting from launch
```

Comment:
Any reason not to expose the timeout conig on this `WithPred` constructor? Is it not necessary on the place where it is called?

Reply:
This is only called from GetOrchestrators on the `DBOrchestratorPoolCache` and with the global variable set at launch it didn't seem like I needed to add it here too.  

Added it with the update to put the discoveryTimeout in the pool objects.